openiddict/src/Audacia.Auth.OpenIddict/Common/UtcTimeProvider.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+#if NET6_0
+using Microsoft.AspNetCore.Authentication;
+#endif
+
+namespace Audacia.Auth.OpenIddict.Common;
+
+/// <summary>
+/// Default <see cref="IUtcTimeProvider"/> implementaiton.
+/// </summary>
+[SuppressMessage("Performance", "CA1812:Avoid uninstantiated internal classes", Justification = "Registered in dependency injection.")]
+internal class UtcTimeProvider : IUtcTimeProvider
```

Comment:
`ISystemClock` is deprecated in .NET 8 in favour of `TimeProvider`, added this abstraction so it was easy to use preprocessor directives to change the implementation. 

This is untested at runtime. I think there needs to be a call to `builder.Services.AddSingleton<TimeProvider>(TimeProvider.System);` to configure this in DI.

Reply:
If the `TimeProvider.System` is added in DI, it will need wrapping in a `#if NET8_0` preprocessor.

Alternatively, we might want to leave injecting a `TimeProvider` up to the consuming code as if someone if injecting their own `TimeProvider` we would end up with two `TimeProviders` in DI. If so, the README will need updating.

Reply:
Does `ISystemClock` currently need to be registered in DI (I can't see it being done anywhere)? Or does .NET automatically register somehow? It would be nice to handle `TimeProvider` however we're currently handling `ISystemClock`, if possible.

Reply:
I think it is usually registered by the `services.AddAuthentication` call - https://github.com/dotnet/aspnetcore/blob/62351067ff4c1401556725b401478e648b66acdc/src/Security/Authentication/Core/src/AuthenticationServiceCollectionExtensions.cs#L26

Reply:
In fact here's the code on the main branch - they are disabling the obsolete warning (which is also an option for us here)
https://github.com/dotnet/aspnetcore/blob/099bc84939fec64e803ec173dac6aa8ceb46a8cc/src/Security/Authentication/Core/src/AuthenticationServiceCollectionExtensions.cs#L28

Reply:
TimeProvider was already registered when Navod tested this on .NET 8.